docs/bugs.json

```diff
@@ -1,4 +1,14 @@
 [
+    {
+        "uid": "SOL-2022-5",
+        "name": "DirtyBytesArrayToStorage",
+        "summary": "Copying ``bytes`` arrays from memory or calldata to storage may result in dirty storage values.",
+        "description": "Copying ``bytes`` arrays from memory or calldata to storage is done in chunks of 32 bytes. Thereby, dirty values in calldata or memory can be written to storage, which may then become observable after a ``.push()`` on the bytes array in storage.",
```

Comment:
```suggestion
        "description": "Copying ``bytes`` arrays from memory or calldata to storage is done in chunks of 32 bytes. Thereby, dirty values in calldata or memory can be written to storage, which may then become observable after a ``.push()`` on the bytes array in storage. This bug affects only the legacy code generation pipeline. The IR-based pipeline always cleans dirty bytes when copying array elements.",
```

Reply:
I wonder if we should start using ``"viaIR": true`` and ``"viaIR": false`` in the condition field...

Reply:
Yeah, sounds like a good idea.

Reply:
Actually ``docs/bugs.rst`` doesn't document even the ``yulOptimizer`` one so far...
Thinking about it, I'm also not sure if we should change that to plain ``optimizer: true`` for the other bug as - since you don't *explicitly* need to enable the yul optimizer anymore these days...

Reply:
Does `optimizer: true` mean "happens only with optimizer" or "happens with optimizer too"?

Reply:
Also, about the new flag, maybe @axic would like to chime in.

Reply:
> Does `optimizer: true` mean "happens only with optimizer" or "happens with optimizer too"?

See ``docs/bugs.rst`` (even though it's apparently incomplete): 
```
conditions
    Conditions that have to be met to trigger the bug. The following
    keys can be used:
    ``optimizer``, Boolean value which
    means that the optimizer has to be switched on to enable the bug.
    ``evmVersion``, a string that indicates which EVM version compiler
    settings trigger the bug. The string can contain comparison
    operators. For example, ``">=constantinople"`` means that the bug
    is present when the EVM version is set to ``constantinople`` or
    later.
    If no conditions are given, assume that the bug is present.
```

I.e. it means happens *only* with optimizer.